### PR TITLE
Added PHP 8.0 support #4

### LINKED ISSUE #4: PHP 8.0 support
### Feature Request

|    Q        |   A
|------------ | ------
| New Feature | yes

#### Summary

To be prepared for the december release of PHP 8.0, this repository has some additional TODOs to be tested against the new major version.

In order to make this repository compatible, one has to follow these steps:

- [ ] Modify `composer.json` to provide support for PHP 8.0 by adding the constraint `~8.0.0`
- [ ] Modify `composer.json` to drop support for PHP less than 7.3
- [ ] Modify `composer.json` to implement phpunit 9.3 which supports PHP 7.3+
- [ ] Modify `.travis.yml` to ignore platform requirements when installing composer dependencies (simply add `--ignore-platform-reqs` to `COMPOSER_ARGS` env variable)
- [ ] Modify `.travis.yml` to add PHP 8.0 to the matrix (*NOTE*: Do not allow failures as PHP 8.0 has a feature freeze since 2020-08-04!)
- [ ] Modify source code in case there are incompatibilities with PHP 8.0

### PR DESCRIPTION
**Pull request by svycka**
Q | A
-- | --
New Feature | yes

Summary
To be prepared for the december release of PHP 8.0, this repository has some additional TODOs to be tested against the new major version.

In order to make this repository compatible, one has to follow these steps:

- [x] Modify composer.json to provide support for PHP 8.0 by adding the constraint ~8.0.0
- [x]  Modify composer.json to drop support for PHP less than 7.3
- [x]  Modify composer.json to implement phpunit 9.3 which supports PHP 7.3+
- [x]  Modify .travis.yml to ignore platform requirements when installing composer dependencies (simply add --ignore-platform-reqs to COMPOSER_ARGS env variable)
- [x]  Modify .travis.yml to add PHP 8.0 to the matrix (NOTE: Do not allow failures as PHP 8.0 has a feature freeze since 2020-08-04!)
- [x]  Modify source code in case there are incompatibilities with PHP 8.0

closes #4

### CODE REVIEW

File: test/LaminasViewRendererTest.php [hunk 1]
```diff
@@ -83,18 +86,25 @@ public function assertEqualTemplatePath(TemplatePath $expected, TemplatePath $re
         }
     }
 
+    public function fetchPhpRenderer(LaminasViewRenderer $view)
+    {
+        $r = new \ReflectionProperty($view, 'renderer');
```
Comment (samsonasik) [inline test/LaminasViewRendererTest.php]: can be moved to use statement

**Comment by svycka** [comment]
Coverage decreased but I can't see why I guess because of PHPUnit and coverage update because it was not covered before but it was reported as covered for some reason as I understand

**Comment by svycka** [comment]
Added test to cover uncovered line, coverage increased but coveralls still complains not sure why :) My guess that it compairs coverage with 2.2.x branch instead of 3.0.x

File: .travis.yml [hunk 2]
```diff
@@ -6,30 +6,30 @@ cache:
 
 env:
   global:
-    - COMPOSER_ARGS="--no-interaction"
+    - COMPOSER_ARGS="--no-interaction --ignore-platform-reqs"
```
Comment (froschdesign) [inline .travis.yml]: The new parameter is only needed for the nightly version. Please move this to the related section.

File: .travis.yml [hunk 3]
```diff
@@ -6,30 +6,30 @@ cache:
...
       env:
         - DEPS=lowest
-    - php: 7.2
+    - php: 7.4
       env:
         - DEPS=latest
-    - php: 7.3
+    - php: nightly
       env:
         - DEPS=lowest
```
Comment (froschdesign) [inline .travis.yml:31]:
```suggestion
    - php: nightly
      env:
        - DEPS=lowest
        - COMPOSER_ARGS="--no-interaction --ignore-platform-reqs"
```

File: .travis.yml [hunk 4]
```diff
@@ -6,30 +6,30 @@ cache:
...
-    - php: 7.3
+    - php: nightly
       env:
         - DEPS=latest
```
Comment (froschdesign) [inline .travis.yml:35]:
```suggestion
    - php: nightly
      env:
        - DEPS=latest
        - COMPOSER_ARGS="--no-interaction --ignore-platform-reqs"
```

File: phpunit.xml.dist [hunk 5]
```diff
@@ -1,13 +1,13 @@
...
-        </testsuite>
-    </testsuites>
-
-    <filter>
-        <whitelist processUncoveredFilesFromWhitelist="true">
-            <directory suffix=".php">src</directory>
-        </whitelist>
-    </filter>
+<?xml version="1.0"?>
+<phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" bootstrap="./vendor/autoload.php" colors="true" xsi:noNamespaceSchemaLocation="https://schema.phpunit.de/9.3/phpunit.xsd">
```
Comment (froschdesign) [inline phpunit.xml.dist]: In the Composer configuration PHPUnit is set to version 9.4.x but the XML schema definition of version 9.3 is used here.
Therefore I would suggest the following:
```suggestion
<phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" bootstrap="./vendor/autoload.php" colors="true" xsi:noNamespaceSchemaLocation="vendor/phpunit/phpunit/phpunit.xsd">
```

File: test/LaminasViewRendererTest.php [hunk 6]
```diff
@@ -31,6 +35,8 @@
 
 class LaminasViewRendererTest extends TestCase
 {
+    use ProphecyTrait;
+
```
Comment (froschdesign) [inline test/LaminasViewRendererTest.php]: Maybe I missed something, but why is this trait necessary here?

**Review by froschdesign — CHANGES_REQUESTED** [review]
A big thank you for your contribution! 👍 

I only added some small comments and suggestions. Thanks in advance!

**Comment by adamturcsan** [comment]
@svycka @froschdesign May I help you somehow to finish this PR?

File: .travis.yml [hunk 7]
```diff
@@ -12,26 +12,28 @@ env:
...
-    - php: 7.2
+    - php: 7.4
       env:
         - DEPS=lowest
-    - php: 7.2
+    - php: 7.4
       env:
         - DEPS=latest
-    - php: 7.3
+    - php: nightly
```
Comment (froschdesign) [inline .travis.yml:29]:
```suggestion
    - php: 8.0
```

File: .travis.yml [hunk 8]
```diff
@@ -12,26 +12,28 @@ env:
...
       env:
         - DEPS=lowest
-    - php: 7.3
+        - COMPOSER_ARGS="--no-interaction --ignore-platform-reqs"
+    - php: nightly
```
Comment (froschdesign) [inline .travis.yml:33]:
```suggestion
    - php: 8.0
```

File: .travis.yml [hunk 9]
```diff
@@ -12,26 +12,28 @@ env:
...
       env:
         - DEPS=latest
+        - COMPOSER_ARGS="--no-interaction --ignore-platform-reqs"
```
Comment (froschdesign) [inline .travis.yml:36]:
```suggestion
```

**Comment by weierophinney** [comment]
Closing in favor of #6, which also updated us to github actions (allowing us to finish validation of testing).

**Pull request closed without merging**